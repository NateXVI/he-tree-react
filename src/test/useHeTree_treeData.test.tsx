
import {
  useHeTree,
} from "../../lib/HeTree";
import renderer from "react-test-renderer";
import { useState } from 'react';
import { createData } from "./treeData.test";

// let tree = toJson(component)
// expect(tree).toMatchSnapshot()

test('render tree', () => {
  // const element = document.createElement('div')
  // expect(element).not.toBeNull()
  const Test = () => {
    const [data, setdata] = useState(createData);
    const { renderHeTree } = useHeTree({
      data,
      dataType: 'tree',
      onChange: setdata,
      renderNode: ({ id, node, open, checked }) => <div>
        {node.name}
      </div>,
    })
    return renderHeTree()
  }
  const component = renderer.create(
    <Test />,
  )
})

export function toJson(component: renderer.ReactTestRenderer) {
  const result = component.toJSON()
  expect(result).toBeDefined()
  expect(result).not.toBeInstanceOf(Array)
  return result as renderer.ReactTestRendererJSON
}