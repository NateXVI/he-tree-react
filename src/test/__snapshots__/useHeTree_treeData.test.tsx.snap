// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`render tree 1`] = `
<div
  className="he-tree "
  onDragEnd={[Function]}
  onDragOver={[Function]}
  onDrop={[Function]}
>
  <div
    onScroll={[Function]}
    style={
      {
        "height": "100%",
        "overflow": "auto",
      }
    }
  >
    <div
      style={
        {
          "display": "flex",
          "flexDirection": "column",
        }
      }
    >
      <div
        data-key="1"
        data-level="1"
        data-node-box={true}
        draggable={true}
        onDragLeave={[Function]}
        onDragStart={[Function]}
        style={
          {
            "paddingLeft": "0px",
          }
        }
      >
        <div>
          Root Category
        </div>
      </div>
      <div
        data-key="2"
        data-level="2"
        data-node-box={true}
        draggable={true}
        onDragLeave={[Function]}
        onDragStart={[Function]}
        style={
          {
            "paddingLeft": "20px",
          }
        }
      >
        <div>
          Technology
        </div>
      </div>
      <div
        data-key="5"
        data-level="3"
        data-node-box={true}
        draggable={true}
        onDragLeave={[Function]}
        onDragStart={[Function]}
        style={
          {
            "paddingLeft": "40px",
          }
        }
      >
        <div>
          Hardware
        </div>
      </div>
      <div
        data-key="10"
        data-level="4"
        data-node-box={true}
        draggable={true}
        onDragLeave={[Function]}
        onDragStart={[Function]}
        style={
          {
            "paddingLeft": "60px",
          }
        }
      >
        <div>
          Computer Components
        </div>
      </div>
      <div
        data-key="4"
        data-level="3"
        data-node-box={true}
        draggable={true}
        onDragLeave={[Function]}
        onDragStart={[Function]}
        style={
          {
            "paddingLeft": "40px",
          }
        }
      >
        <div>
          Programming
        </div>
      </div>
      <div
        data-key="8"
        data-level="4"
        data-node-box={true}
        draggable={true}
        onDragLeave={[Function]}
        onDragStart={[Function]}
        style={
          {
            "paddingLeft": "60px",
          }
        }
      >
        <div>
          Python
        </div>
      </div>
      <div
        data-key="3"
        data-level="2"
        data-node-box={true}
        draggable={true}
        onDragLeave={[Function]}
        onDragStart={[Function]}
        style={
          {
            "paddingLeft": "20px",
          }
        }
      >
        <div>
          Science
        </div>
      </div>
      <div
        data-key="7"
        data-level="3"
        data-node-box={true}
        draggable={true}
        onDragLeave={[Function]}
        onDragStart={[Function]}
        style={
          {
            "paddingLeft": "40px",
          }
        }
      >
        <div>
          Biology
        </div>
      </div>
      <div
        data-key="6"
        data-level="3"
        data-node-box={true}
        draggable={true}
        onDragLeave={[Function]}
        onDragStart={[Function]}
        style={
          {
            "paddingLeft": "40px",
          }
        }
      >
        <div>
          Physics
        </div>
      </div>
    </div>
  </div>
</div>
`;
